### DISCUSSION TITLE
CalendarDatePicker could not display Chinese characters properly in browser

### DISCUSSION THREAD

**Discussion by dirdir** [AvaloniaUI/Avalonia#17625] [Q&A] [answered]
![CalendarDatePicker2](https://github.com/user-attachments/assets/ebd21359-5332-4b1c-8820-88a8c5bb39db)
CalendarDatePicker cannot display Chinese properly in the browser project, the related TTF font files have been added to the resource, and FontFamily has been set

**Comment by rabbitism** [comment]
Can you please provide two more test cases and check if chinese characters show up?
1 place a calendar in your main layout.
2 the context menu of textbox.

**Reply by dirdir** [reply]
![image](https://github.com/user-attachments/assets/2f11e813-f3d1-41ae-9596-03531cf7281d)
Thanks for your reply. The Chinese characters on the Calendar control look normal, while the context menu displays English.

**Reply by rabbitism** [reply]
>Thanks for your reply. The Chinese characters on the Calendar control look normal, while the context menu displays English.

Looks like font is not property assigned to your popups.

**Reply by Gillibald** [reply]
You should define font fallbacks for the missing unicode ranges. That way the font manager will be able to find a fallbacks for missing glyphs. Directly assigning a specific font family isn't recommended.

**Reply by dirdir** [reply]
> You should define font fallbacks for the missing unicode ranges. That way the font manager will be able to find a fallbacks for missing glyphs. Directly assigning a specific font family isn't recommended.

It seems to make sense, but I don't know what to do now. Thanks!

**Reply by Gillibald** [reply, marked as answer]
https://github.com/AvaloniaUI/Avalonia/pull/7089
The default UnicodeRange covers all codepoints. As long as you specify a FontFamily via the FontFallback API everything should render on WASM
